internal/connectors/plugins/plugin.go

```diff
@@ -27,13 +32,41 @@ func NewPlugin(name string, pluginConstructorFn models.PluginConstructorFn) *cob
 	return cmd
 }
 
-func runServer(pluginConstructorFn models.PluginConstructorFn) func(cmd *cobra.Command, args []string) error {
+func runServer(
+	pluginConstructorFn models.PluginConstructorFn,
+) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		// TODO initialise logger here?
-		opts := fx.Options(
-			fx.Provide(pluginConstructorFn, NewServer),
+		hlogger := hclog.New(loggerOptions())
+		hclog.SetDefault(hlogger)
+
+		opts := make([]fx.Option, 0)
+		opts = append(opts,
+			otlp.FXModuleFromFlags(cmd),
+			otlpmetrics.FXModuleFromFlags(cmd),
+			fx.Decorate(fx.Annotate(func(meterProvider metric.MeterProvider) (metrics.MetricsRegistry, error) {
+				hlogger.Info("decorate metrics registery called")
+				return metrics.RegisterMetricsRegistry(meterProvider)
+			})),
+			fx.Provide(metrics.RegisterMetricsRegistry),
+			fx.Invoke(func(metrics.MetricsRegistry) {}),
```

Comment:
why the invoke here without anything in it ?

Reply:
It was to force `RegisterMetricsRegistry()` to be called. Otherwise we were getting a `noop.MetricsRegistry` because `metrics.MetricsRegistry` isn't provided to any of the other constructors.

Maybe there is a more colloquial way to do it though?